### PR TITLE
[YUNIKORN-2233] Scheduler core cannot be stopped properly

### PR DESCRIPTION
**Pull request by pbacsko**
### What is this PR for?
After calling `MockScheduler.Stop()`, we still have a bunch of goroutines running. This is not a problem on a real cluster, because a restart means a complete shutdown of the Yunikorn process. During testing, this leaves garbage behind and generates possible unwanted interactions between test cases.

### What type of PR is it?
* [ ] - Bug Fix
* [x] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
* [ ] - Task

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-2233

### How should this be tested?
Insert a breakpoint after `ms.Stop()` in a test which uses the Mock scheduler. Check the running goroutines with the debugger.

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

File: pkg/entrypoint/service_context.go [hunk 1]
```diff
@@ -21,25 +21,36 @@ package entrypoint
...
 		if err := s.WebApp.StopWebApp(); err != nil {
 			log.Log(log.Entrypoint).Error("failed to stop web-app",
 				zap.Error(err))
 		}
 	}
+	if s.MetricsCollector != nil {
+		s.MetricsCollector.Stop()
+	}
+	s.Scheduler.Stop()
+	if proxyImpl, ok := s.RMProxy.(*rmproxy.RMProxy); ok {
```
Comment (pbacsko) [inline pkg/entrypoint/service_context.go:52]: NOTE: Currently there's no Stop() method on  `api.SchedulerAPI`.

Comment (wilfred-s) [reply pkg/entrypoint/service_context.go:52]: change committed in SI

**Review by craigcondit — APPROVED** [review]
+1 LGTM.

**Review by wilfred-s — APPROVED** [review]
LGTM
nice cleanup for the stops

**Pull request closed without merging**